### PR TITLE
*: Add ability to set memlock rlimit

### PR DESCRIPTION
**Pull request by kakkoyun**
Signed-off-by: Kemal Akkoyun <kakkoyun@gmail.com>

cc @javierhonduco

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -66,6 +66,10 @@ Flags:
                                   the Kubernetes node name.
       --config-path="parca-agent.yaml"
                                   Path to config file.
+      --memlock-rlimit=4294967296
+                                  The value for the RLimit for memlock. Used
```
Comment (javierhonduco) [inline README.md:70]: nit: Should we specify the unit?

File: cmd/parca-agent/main.go [hunk 2]
```diff
@@ -72,13 +73,17 @@ var (
...
 type flags struct {
 	LogLevel    string `kong:"enum='error,warn,info,debug',help='Log level.',default='info'"`
 	HTTPAddress string `kong:"help='Address to bind HTTP server to.',default=':7071'"`
 
-	Node string `kong:"help='The name of the node that the process is running on. If on Kubernetes, this must match the Kubernetes node name.',default='${hostname}'"`
-
-	ConfigPath string `default:"parca-agent.yaml" help:"Path to config file."`
+	Node          string `kong:"help='The name of the node that the process is running on. If on Kubernetes, this must match the Kubernetes node name.',default='${hostname}'"`
+	ConfigPath    string `default:"parca-agent.yaml" help:"Path to config file."`
+	MemlockRlimit uint64 `default:"${default_memlock_rlimit}" help:"The value for the RLimit for memlock. Used to ensure the agent can lock memory for eBPF maps. 0 means no limit."`
```
Comment (javierhonduco) [inline cmd/parca-agent/main.go:86]: > ${default_memlock_rlimit}"

What! TIL! :D

File: pkg/profiler/rlimit.go [hunk 3]
```diff
@@ -16,22 +16,35 @@ package profiler
...
+	}
+	if max == 0 {
+		max = unix.RLIM_INFINITY
+	}
 	rLimit := syscall.Rlimit{
 		Cur: cur,
 		Max: max,
 	}
+
+	rlimitMu.Lock()
```
Comment (javierhonduco) [inline pkg/profiler/rlimit.go:41]: Why is this needed?

We don't call it concurrently no? Either way, it seems like it's thread-safe:

>         │getrlimit(), setrlimit(), prlimit()   │ Thread safety │ MT-Safe │

https://man7.org/linux/man-pages/man2/getrlimit.2.html

**Review by javierhonduco — APPROVED** [review]
Looks great!! 💯

File: README.md (hunk 1 above)
Comment (kakkoyun) [reply README.md:70]: Good catch!

File: pkg/profiler/rlimit.go (hunk 3 above)
Comment (kakkoyun) [reply pkg/profiler/rlimit.go:41]: It's primarily future envy. We will have more than one profiler eventually, and this API is designed to be called from the profilers. Happy to get rid of it.
